utils/orchestrators.ts

```diff
@@ -0,0 +1,120 @@
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import {
+  IResponseErrorConflict,
+  IResponseErrorInternal,
+  IResponseSuccessAccepted,
+  ResponseErrorConflict,
+  ResponseErrorInternal,
+  ResponseSuccessAccepted
+} from "italia-ts-commons/lib/responses";
+
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import { PromiseType } from "italia-ts-commons/lib/types";
+import { StatusEnum as CgnActivatedStatusEnum } from "../generated/definitions/CgnActivatedStatus";
+import { StatusEnum as CgnCanceledStatusEnum } from "../generated/definitions/CgnCanceledStatus";
+import { StatusEnum as CgnPendingStatusEnum } from "../generated/definitions/CgnPendingStatus";
+import { StatusEnum as CgnRevokedStatusEnum } from "../generated/definitions/CgnRevokedStatus";
+import { CgnStatus } from "../generated/definitions/CgnStatus";
+
+/**
+ * The identifier for StartEligibilityCheckOrchestrator
+ * @param fiscalCode the id of the requesting user
+ */
+export const makeUpdateCgnOrchestratorId = (
+  fiscalCode: FiscalCode,
+  cgnStatus: string
+) => `${fiscalCode}-UPDCGN-${cgnStatus}`;
+
+/**
+ * Returns the status of the orchestrator augmented with an isRunning attribute
+ */
+export const isOrchestratorRunning = (
+  client: DurableOrchestrationClient,
+  orchestratorId: string
+): TaskEither<
+  Error,
+  PromiseType<ReturnType<typeof client["getStatus"]>> & {
+    isRunning: boolean;
+  }
+> =>
+  tryCatch(() => client.getStatus(orchestratorId), toError).map(status => ({
+    ...status,
+    isRunning:
+      status.runtimeStatus === df.OrchestrationRuntimeStatus.Running ||
+      status.runtimeStatus === df.OrchestrationRuntimeStatus.Pending
+  }));
+
+const cgnStatuses: ReadonlyArray<string> = [
+  CgnRevokedStatusEnum.REVOKED.toString(),
+  CgnActivatedStatusEnum.ACTIVATED.toString(),
+  CgnCanceledStatusEnum.CANCELED.toString(),
+  CgnPendingStatusEnum.PENDING.toString()
+];
+
+export type CheckUpdateCgnIsRunningErrorTypes =
+  | IResponseErrorInternal
+  | IResponseSuccessAccepted
+  | IResponseErrorConflict;
+/**
+ * Check if the current user has a pending cgn status update process.
+ */
+export const checkUpdateCgnIsRunning = (
+  client: DurableOrchestrationClient,
+  fiscalCode: FiscalCode,
+  cgnStatus: CgnStatus
+): TaskEither<CheckUpdateCgnIsRunningErrorTypes, false> =>
+  isOrchestratorRunning(
+    client,
+    makeUpdateCgnOrchestratorId(fiscalCode, cgnStatus.status)
+  )
+    .foldTaskEither<CheckUpdateCgnIsRunningErrorTypes, false>(
+      err =>
+        fromLeft(
+          ResponseErrorInternal(
+            `Error checking UpdateCgnOrchestrator: ${err.message}`
+          )
+        ),
+      ({ isRunning }) =>
+        isRunning ? fromLeft(ResponseSuccessAccepted()) : taskEither.of(false)
+    )
+    .chain(_ =>
+      taskEither.of(
+        cgnStatuses.filter(el => el !== cgnStatus.status.toString())
+      )
+    )
+    .chain(otherStatuses =>
+      array.sequence(taskEither)(
```

Comment:
can you add a comment on what is happening here?

Reply:
Added in [3433895](https://github.com/pagopa/io-functions-cgn/pull/10/commits/34338952fcc66c7f8def966d97b181f70521f1b6)